<?xml version='1.0' encoding='utf-8'?>

<launch>
  <node pkg="xmate_robot_2" type="place_grasp_demo" name="place_grasp_demo" output="screen" >

    <!--抓取物料设置, M8: 1;M10: 2-->
    <param name="target_type" value="2"/>
    <!--螺丝放置位置对应Ar码号-->
    <param name="LS_ID" value="8"/>
    <!--螺母放置位置对应Ar码号-->
    <param name="LM_ID" value="6"/>
    <!--钣金件放置位置对应Ar码号-->
    <param name="BJJ_ID" value="2"/>

    <!--设置机械臂运行速度-->
    <param name="move_sped" value="0.2"/>

    <!-- 中转台抓取物料参数 -->
    <!-- 抓取物料时机械臂的x坐标偏移量(笛卡尔坐标系)，增大往左(默认：0.018) -->
    <param name="grasp_x" value="0.019"/>　<!-- 单位：m-->
    <!-- 抓取物料时机械臂的y坐标偏移量(笛卡尔坐标系)，增大向前(默认：0.052) -->
    <param name="grasp_y" value="0.069"/>　<!-- 单位：m-->
    <!-- 抓取物料时机械臂的z坐标偏移量(笛卡尔坐标系)，往下减小(默认：0.198) -->
    <param name="grasp_z" value="0.20"/>　<!-- 单位：m-->

    <!-- 中转台物料id2在物料盘上面放置参数 -->
    <!-- 抓取id2物料到物料盘时机械臂的x坐标偏移量(笛卡尔坐标系)，增大往前(默认：0.0025) -->
    <param name="graspid2_x" value="0.009"/>　<!-- 单位：m-->
    <!-- 抓取id2物料到物料盘时机械臂的y坐标偏移量(笛卡尔坐标系)，增大向左(默认：-0.000) -->
    <param name="graspid2_y" value="-0.219"/>　<!-- 单位：m-->
    <!-- 抓取id2物料到物料盘时机械臂的z坐标偏移量(笛卡尔坐标系)，增大往下(默认：0.050) -->
    <param name="graspid2_z" value="0.050"/>　<!-- 单位：m-->

    <!-- 中转台物料id1在物料盘上面放置参数 -->
    <!-- 抓取id1物料到物料盘时机械臂的x坐标偏移量(笛卡尔坐标系)，增大往前(默认：-0.0005) -->
    <param name="graspid1_x" value="0.008"/>　<!-- 单位：m-->
    <!-- 抓取id1物料到物料盘时机械臂的y坐标偏移量(笛卡尔坐标系)，增大向左默认：0.109) -->
    <param name="graspid1_y" value="0.0025"/>　<!-- 单位：m-->
    <!-- 抓取id1物料到物料盘时机械臂的z坐标偏移量(笛卡尔坐标系)，增大往下(默认：0.050) -->
    <param name="graspid1_z" value="0.050"/>　<!-- 单位：m-->
   
    <!--中转台抓取钣金件参数-->
    <!-- 抓取物料时机械臂的x坐标偏移量(笛卡尔坐标系)，增大往左(默认：0.018) -->
    <param name="bjj_grasp_x" value="0.0260"/>　<!-- 单位：m-->
    <!-- 抓取物料时机械臂的y坐标偏移量(笛卡尔坐标系)，增大向前(默认：0.052) -->
    <param name="bjj_grasp_y" value="0.056"/>　<!-- 单位：m-->
    <!-- 抓取物料时机械臂的z坐标偏移量(笛卡尔坐标系)，往下减小(默认：0.198) -->
    <param name="bjj_grasp_z" value="0.205"/>　<!-- 单位：m-->

    <!-- 中转台物料id1在物料盘上面放置参数 -->
    <!-- 抓取id1物料到物料盘时机械臂的x坐标偏移量(笛卡尔坐标系)，增大往前(默认：0.0025) -->
    <param name="bjj_place_x" value="-0.0408"/>　<!-- 单位：m-->
    <!-- 抓取id1物料到物料盘时机械臂的y坐标偏移量(笛卡尔坐标系)，增大向左(默认：-0.000) -->
    <param name="bjj_place_y" value="-0.11"/>　<!-- 单位：m-->
    <!-- 抓取id1物料到物料盘时机械臂的z坐标偏移量(笛卡尔坐标系)，增大往下(默认：0.050) -->
    <param name="bjj_place_z" value="0.030"/>　<!-- 单位：m-->

   <!--***************************放置参数****************************-->

     <!-- 物料id1从物料盘上面抓取参数 -->
    <!-- 从物料盘抓取id1物料时机械臂的x坐标偏移量(笛卡尔坐标系)，减小向前(默认：0.015) -->
    <param name="placeid1_x" value="-0.005"/>　<!-- 单位：m-->
    <!-- 从物料盘抓取id1物料时机械臂的y坐标偏移量(笛卡尔坐标系)，增大向左(默认：0.004) -->
    <param name="placeid1_y" value="-0.22"/>　<!-- 单位：m-->
    <!-- 从物料盘抓取id1物料时机械臂的z坐标偏移量(笛卡尔坐标系)，减小往上(默认：0.040134663) -->
    <param name="placeid1_z" value="0.063"/>　<!-- 单位：m-->

    <!-- 物料id2螺母从物料盘上面抓取参数 -->
    <!-- 从物料盘抓取id2物料时机械臂的x坐标偏移量(笛卡尔坐标系)，减小向前(默认：0.0) -->
    <param name="placeid2_x" value="-0.005"/>　<!-- 单位：m-->
    <!-- 从物料盘抓取id2物料时机械臂的y坐标偏移量(笛卡尔坐标系)，增大向左(默认：0.1115) -->
    <param name="placeid2_y" value="0.0025"/>　<!-- 单位：m-->
    <!-- 从物料盘抓取id2物料时机械臂的z坐标偏移量(笛卡尔坐标系)，往下减小(默认：0.041234663) -->
    <param name="placeid2_z" value="0.063"/>　<!-- 单位：m-->

    <!-- 钣金件从物料盘上面抓取参数 -->
    <!-- 从物料盘抓取id2物料时机械臂的x坐标偏移量(笛卡尔坐标系)，增大往后(默认：0.0) -->
    <param name="zpt_bjj_grasp_x" value="0.0430802"/>　<!-- 单位：m-->
    <!-- 从物料盘抓取id2物料时机械臂的y坐标偏移量(笛卡尔坐标系)，增大向左(默认：0.1115) -->
    <param name="zpt_bjj_grasp_y" value="-0.109"/>　<!-- 单位：m-->
    <!-- 从物料盘抓取id2物料时机械臂的z坐标偏移量(笛卡尔坐标系)，增大往下(默认：0.041234663) -->
    <param name="zpt_bjj_grasp_z" value="0.0420"/>　<!-- 单位：m-->
    
     <!-- 钣金件放到装配台上面放置参数 -->
    <!-- 从物料盘抓取id2物料时机械臂的x坐标偏移量(笛卡尔坐标系)，增大往左(默认：0.0) -->
    <param name="zpt_bjj_place_x" value="0.018"/>　<!-- 单位：m-->
    <!-- 从物料盘抓取id2物料时机械臂的y坐标偏移量(笛卡尔坐标系)，增大向前(默认：0.1115) -->
    <param name="zpt_bjj_place_y" value="0.053"/>　<!-- 单位：m-->
    <!-- 从物料盘抓取id2物料时机械臂的z坐标偏移量(笛卡尔坐标系)，往下减小(默认：0.041234663) -->
    <param name="zpt_bjj_place_z" value="0.216"/>　<!-- 单位：m-->


    <!-- 设置放置物料参数 -->
    <!-- 放置物料到装配台时机械臂的x坐标偏移量(笛卡尔坐标系)，增大往左(默认：0.020) -->
    <param name="place_x" value="0.020"/>　<!-- 单位：m-->
    <!-- 放置物料到装配台时机械臂的y坐标偏移量(笛卡尔坐标系)，增大向前(默认：0.032) -->
    <param name="place_y" value="0.0445"/>　<!-- 单位：m-->
    <!-- 放置物料到装配台时机械臂的z坐标偏移量(笛卡尔坐标系)，往下增大(默认：0.228) -->  
    <param name="place_z" value="0.216"/>　<!-- 单位：m-->
   
   <!--***************************成品抓取放置参数****************************-->
    
    <!-- 成品从装配台上面抓取参数 -->
    <!-- 从物料盘抓取id2物料时机械臂的x坐标偏移量(笛卡尔坐标系)，增大向后(默认：0.0) -->
    <param name="cp_bjj_grasp_x" value="0.0410802"/>　<!-- 单位：m-->
    <!-- 从物料盘抓取id2物料时机械臂的y坐标偏移量(笛卡尔坐标系)，增大向左(默认：0.1115) -->
    <param name="cp_bjj_grasp_y" value="-0.109"/>　<!-- 单位：m-->
    <!-- 从物料盘抓取id2物料时机械臂的z坐标偏移量(笛卡尔坐标系)，增大向下(默认：0.041234663) -->
    <param name="cp_bjj_grasp_z" value="0.0420"/>　<!-- 单位：m-->
    
     <!-- 成品放到AGV上面放置参数 -->
    <!-- 从物料盘抓取id2物料时机械臂的x坐标偏移量(笛卡尔坐标系)，增大向左(默认：0.0) -->
    <param name="cp_bjj_place_x" value="0.016"/>　<!-- 单位：m-->
    <!-- 从物料盘抓取id2物料时机械臂的y坐标偏移量(笛卡尔坐标系)，增大向前(默认：0.1115) -->
    <param name="cp_bjj_place_y" value="0.050"/>　<!-- 单位：m 0.0945-->
    <!-- 从物料盘抓取id2物料时机械臂的z坐标偏移量(笛卡尔坐标系)，增大向上(默认：0.041234663) -->
    <param name="cp_bjj_place_z" value="0.208"/>　<!-- 单位：m-->

    <!-- 从物料盘上抓取成品 -->
    <param name="product_grasp_x" value="0.0430802"/>　<!-- 单位：m-->　　<!--增大向后-->
    <param name="product_grasp_y" value="-0.112"/>　<!-- 单位：m-->　　　<!--增大向左-->
    <param name="product_grasp_z" value="0.042"/>　<!-- 单位：m-->　　<!--增大向下-->

    <!-- 将成品放置到立体仓库 -->
    <param name="Ste_warehouse_x" value="0.02"/>　<!-- 单位：m-->　　<!--增大向右-->
    <param name="Ste_warehouse_y" value="0.03"/>　<!-- 单位：m-->　　<!--增大向前-->
    <param name="Ste_warehouse_z" value="0.004"/>　<!-- 单位：m-->　　<!--增大向上-->

    <!--放置到立体仓库的工位-->
    <param name="arcodeid" value="2"/>

  </node>

</launch>
